cytominer_eval/tests/test_operations/test_mp_value.py

```diff
@@ -0,0 +1,88 @@
+import os
+import pytest
+import pathlib
+import tempfile
+import numpy as np
+import pandas as pd
+from math import isclose
+from cytominer_eval.operations import mp_value
+from cytominer_eval.operations.util import calculate_mp_value, calculate_mahalanobis
+
+# Load CRISPR dataset
+example_file = "SQ00014610_normalized_feature_select.csv.gz"
+example_file = pathlib.Path(
+    "{file}/../../example_data/gene/{eg}".format(
+        file=os.path.dirname(__file__), eg=example_file
+    )
+)
+
+df = pd.read_csv(example_file)
+
+meta_features = [
+    x for x in df.columns if (x.startswith("Metadata_") or x.startswith("Image_"))
+]
+features = df.drop(meta_features, axis="columns").columns.tolist()
+
+control_perts = ["Luc-2", "LacZ-2", "LacZ-3"]
+replicate_id = "Metadata_pert_name"
+
+
+def test_calculate_mahalanobis():
+    sub_df = df[(df.Metadata_WellRow == "A") & (df.Metadata_pert_name == "EMPTY")][
+        features
+    ]
+    control_df = df[df[replicate_id].isin(control_perts)][features]
+
+    maha = calculate_mahalanobis(pert_df=sub_df, control_df=control_df)
+
+    assert isinstance(maha, float)
+    # NB: the following value is empirically determined
+    # and not theoretically justified but avoids unwanted
+    # changes in the implementation of the Mahalanobis distance
+    assert isclose(maha, 3.62523778789, abs_tol=1e-09)
+
+    maha = calculate_mahalanobis(pert_df=control_df, control_df=control_df)
+
+    # Distance to itself should be approximately zero
+    assert isclose(maha, 0, abs_tol=1e-05)
+
+
+def test_calculate_mp_value():
+    # The mp-values are empirical p-values
+    # so they range from 0 to 1, with low values
+    # showing a difference to the control condition.
+
+    sub_df = df[(df.Metadata_WellRow == "A") & (df.Metadata_pert_name == "EMPTY")][
+        features
+    ]
+    control_df = df[df[replicate_id].isin(control_perts)][features]
+
+    # Avoid fluctuations in permutations
+    np.random.seed(2020)
+    result = calculate_mp_value(pert_df=sub_df, control_df=control_df)
+
+    assert isinstance(result, float)
+    assert result > 0
+    assert result < 1
+
+    # Distance to itself should be approximately zero
+    # So mp-value should be 1
+    result = calculate_mp_value(
+        pert_df=control_df, control_df=control_df, params={"nb_permutations": 2000}
+    )
+
+    assert isclose(result, 1, abs_tol=1e-02)
+
+
+def test_mp_value():
+    result = mp_value(
+        df=df,
+        control_perts=control_perts,
+        replicate_id=replicate_id,
+        features=features,
+    )
+
+    assert "mp_value" in result.columns
+    assert all(result.mp_value <= 1)
+    assert all(result.mp_value >= 0)
+    assert len(np.unique(df[replicate_id])) == len(result)
```

Comment:
can you add one more test in `calculate_mp_value()`?

We should enter the following statement:

```python
    assert all(
        [x in p.keys() for x in params.keys()]
    ), "Unknown parameters provided. Only {e} are supported.".format(e=p.keys())
    for (k, v) in params.items():
        p[k] = v
```

I think something like this should work:

```python
with pytest.raises(AssertionError) as ae:
    result = mp_value(
        df=df,
        control_perts=control_perts,
        replicate_id=replicate_id,
        features=features,
    )
    assert "Unknown parameters provided. Only" in str(ae.value)
```

but wait! I just realized we won't ever have access to changing parameters since `mp_value()` doesn't include this option in https://github.com/cytomining/cytominer-eval/pull/26/files#diff-470d83233c24bd0089343f7748832f6036c3b7e8422e1aaaebed657fa7c6a725R46.

Ok, one important question then: Do you think it is important to be able to change these parameters?

If not, then I might prefer to add this as a github issue for a future enhancement. If so, then we should at the very least propagate the change to `mp_value()`, and if super duper important, then we need to propagate the change all the way up to `evaluate()`.